# ---------------------------- #
# JOINT TRAJECTORY CONTROLLERS #
# ---------------------------- #
# gazebo_ros_control:   
#   pid_gains:
#     mico_joint_1:
#       p: 100.0
#       i: 0.01 
#       d: 10.0
#     mico_joint_2:
#       p: 10.0
#       i: 0.01 
#       d: 10.0
#     mico_joint_3:
#       p: 10.0
#       i: 0.01 
#       d: 10.0

mico:
    velocity_finger_trajectory_controller:
        constraints:
          goal_time: 1.0
          mico_joint_finger_1:
            goal: 0.02
            trajectory: 0.05
          mico_joint_finger_2:
            goal: 0.02
            trajectory: 0.05
          stopped_velocity_tolerance: 0.02
        gains:
          mico_joint_finger_1:
            d: 0
            i: 0
            i_clamp: 1
            p: 10
          mico_joint_finger_2:
            d: 0
            i: 0
            i_clamp: 1
            p: 10
        joints:
          - mico_joint_finger_1
          - mico_joint_finger_2
        type: velocity_controllers/JointTrajectoryController
    
    velocity_joint_trajectory_controller:
        constraints:
          goal_time: 1.0
          mico_joint_1:
            goal: 0.02
            trajectory: 0.05
          mico_joint_2:
            goal: 0.02
            trajectory: 0.05
          mico_joint_3:
            goal: 0.02
            trajectory: 0.05
          mico_joint_4:
            goal: 0.02
            trajectory: 0.05
          mico_joint_5:
            goal: 0.02
            trajectory: 0.05
          mico_joint_6:
            goal: 0.02
            trajectory: 0.05
          stopped_velocity_tolerance: 0.02
        gains:
          mico_joint_1:
            d: 5
            i: 10
            i_clamp: 10
            p: 500
          mico_joint_2:
            d: 5
            i: 10
            i_clamp: 10
            p: 500
          mico_joint_3:
            d: 5
            i: 10
            i_clamp: 5
            p: 500
          mico_joint_4:
            d: 5
            i: 10
            i_clamp: 5
            p: 300
          mico_joint_5:
            d: 5
            i: 10
            i_clamp: 5
            p: 300
          mico_joint_6:
            d: 5
            i: 10
            i_clamp: 5
            p: 300
        joints:
        - mico_joint_1
        - mico_joint_2
        - mico_joint_3
        - mico_joint_4
        - mico_joint_5
        - mico_joint_6
        type: velocity_controllers/JointTrajectoryController
    
    joint_state_controller:
      publish_rate: 50
      type: joint_state_controller/JointStateController



effort_finger_trajectory_controller:
    constraints:
      goal_time: 1.0
      mico_joint_finger_1:
        goal: 0.02
        trajectory: 0.05
      mico_joint_finger_2:
        goal: 0.02
        trajectory: 0.05
      stopped_velocity_tolerance: 0.02
    gains:
      mico_joint_finger_1:
        d: 0
        i: 2
        i_clamp: 1
        p: 20
      mico_joint_finger_2:
        d: 0
        i: 2
        i_clamp: 1
        p: 20
    joints:
    - mico_joint_finger_1
    - mico_joint_finger_2
    type: effort_controllers/JointTrajectoryController

effort_joint_trajectory_controller:
    constraints:
      goal_time: 1.0
      mico_joint_1:
        goal: 0.02
        trajectory: 0.05
      mico_joint_2:
        goal: 0.02
        trajectory: 0.05
      mico_joint_3:
        goal: 0.02
        trajectory: 0.05
      mico_joint_4:
        goal: 0.02
        trajectory: 0.05
      mico_joint_5:
        goal: 0.02
        trajectory: 0.05
      mico_joint_6:
        goal: 0.02
        trajectory: 0.05
      stopped_velocity_tolerance: 0.02
    gains:
      mico_joint_1:
        d: 0
        i: 0
        i_clamp: 10
        p: 5000
      mico_joint_2:
        d: 0
        i: 0
        i_clamp: 10
        p: 5000
      mico_joint_3:
        d: 0
        i: 0
        i_clamp: 10
        p: 5000
      mico_joint_4:
        d: 0
        i: 0
        i_clamp: 10
        p: 500
      mico_joint_5:
        d: 0
        i: 0
        i_clamp: 10
        p: 200
      mico_joint_6:
        d: 0
        i: 0
        i_clamp: 10
        p: 500
    joints:
    - mico_joint_1
    - mico_joint_2
    - mico_joint_3
    - mico_joint_4
    - mico_joint_5
    - mico_joint_6
    type: effort_controllers/JointTrajectoryController

# -------------------- #
# POSITION CONTROLLERS #
# -------------------- #

finger_1_position_controller:
    joint: mico_joint_finger_1
    pid:
      d: 0
      i: 0
      p: 10
    type: velocity_controllers/JointPositionController

finger_2_position_controller:
    joint: mico_joint_finger_2
    pid:
      d: 0
      i: 0
      p: 10
    type: velocity_controllers/JointPositionController

joint_1_position_controller:
    joint: mico_joint_1
    pid:
      d: 5
      i: 2
      p: 500
    type: velocity_controllers/JointPositionController

joint_2_position_controller:
    joint: mico_joint_2
    pid:
      d: 5
      i: 2
      p: 500
    type: velocity_controllers/JointPositionController

joint_3_position_controller:
    joint: mico_joint_3
    pid:
      d: 5
      i: 2
      p: 100
    type: velocity_controllers/JointPositionController

joint_4_position_controller:
    joint: mico_joint_4
    pid:
      d: 0
      i: 2
      p: 100
    type: velocity_controllers/JointPositionController

joint_5_position_controller:
    joint: mico_joint_5
    pid:
      d: 0
      i: 0
      p: 50
    type: velocity_controllers/JointPositionController

joint_6_position_controller:
    joint: mico_joint_6
    pid:
      d: 0
      i: 2
      p: 100
    type: velocity_controllers/JointPositionController


# -------------------- #
# VELOCITY CONTROLLERS #
# -------------------- #

finger_1_velocity_controller:
    joint: mico_joint_finger_1
    pid:
      d: 0
      i: 0
      p: 10
    type: effort_controllers/JointVelocityController

finger_2_velocity_controller:
    joint: mico_joint_finger_2
    pid:
      d: 0
      i: 0
      p: 10
    type: effort_controllers/JointVelocityController

joint_1_velocity_controller:
    joint: mico_joint_1
    pid:
      d: 0
      i: 0
      p: 5000
    type: effort_controllers/JointVelocityController

joint_2_velocity_controller:
    joint: mico_joint_2
    pid:
      d: 0
      i: 0
      p: 5000
    type: effort_controllers/JointVelocityController

joint_3_velocity_controller:
    joint: mico_joint_3
    pid:
      d: 0
      i: 0
      p: 5000
    type: effort_controllers/JointVelocityController

joint_4_velocity_controller:
    joint: mico_joint_4
    pid:
      d: 0
      i: 0
      p: 500
    type: effort_controllers/JointVelocityController

joint_5_velocity_controller:
    joint: mico_joint_5
    pid:
      d: 0
      i: 0
      p: 200
    type: effort_controllers/JointVelocityController

joint_6_velocity_controller:
    joint: mico_joint_6
    pid:
      d: 0
      i: 0
      p: 500
    type: effort_controllers/JointVelocityController




# -------------------- #
#   EFFORT CONTROLLERS #
# -------------------- #

joint_1_effort_controller:
    joint: mico_joint_1
    type: effort_controllers/JointEffortController

joint_2_effort_controller:
    joint: mico_joint_2
    type: effort_controllers/JointEffortController

joint_3_effort_controller:
    joint: mico_joint_3
    type: effort_controllers/JointEffortController

joint_4_effort_controller:
    joint: mico_joint_4
    type: effort_controllers/JointEffortController

joint_5_effort_controller:
    joint: mico_joint_5
    type: effort_controllers/JointEffortController

joint_6_effort_controller:
    joint: mico_joint_6
    type: effort_controllers/JointEffortController




joint_state_controller:
    publish_rate: 50
    type: joint_state_controller/JointStateController
